source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -472,11 +436,90 @@ void Filter::onComplete(Filters::Common::ExtAuthz::ResponsePtr&& response) {
           config_->statusOnError(), EMPTY_STRING, nullptr, absl::nullopt,
           Filters::Common::ExtAuthz::ResponseCodeDetails::get().AuthzError);
     }
+
     break;
   }
   }
 }
 
+// TODO(leonti): add denylist for header mutation keys.
+void Filter::applyHeaderMutations(const Filters::Common::ExtAuthz::Response& response) {
+  ENVOY_STREAM_LOG(trace, "ext_authz filter added header(s) to the request:", *decoder_callbacks_);
+  for (const auto& header : response.headers_to_set) {
+    if (headerPairIsValid(header)) {
```

Comment:
Is adding headerPairIsValid() to different type of the headers, like headers_to_set, headers_to_add, headers_to_append, headers_to_remove the main change of the PR? And everything else is just refactoring to make it a center place to do this?